### ISSUE TITLE
Slow video playback when connected to docking station

### ISSUE THREAD

**Issue by UnoSD** [DisplayLink/evdi#211]
Initially please make sure the problem hasn't been already addressed:
  * [ Y ] Are you using the latest driver?
    (https://www.displaylink.com/downloads/ubuntu)
  * [ Y ] Are you using the latest EVDI version?
    (https://github.com/DisplayLink/evdi/releases)
  * [ Y ] If you are using a DisplayLink device, have you checked 'troubleshooting'
    on DisplayLink's website?
    (https://support.displaylink.com/knowledgebase/topics/103927-troubleshooting-ubuntu)
  * [ Y ] Is this issue related to evdi/kernel?
    (if it is rather connected to DisplayLinkManager please take a look at support
    https://support.displaylink.com or forum https://www.displaylink.org/forum/)

Some information for baseline would help a lot:
  * Linux distribution and its version
Arch Linux
  * Linux kernel version
5.4.49-1-lts
  * Xorg version (if used)
1.20.8-2
  * Desktop environment in use
XfCE

I tried upgrading to the latest kernel to see if the issues with the crashes were solved and they weren't; in addition, all the videos played using the GPU were at 1FPS.

I then downgraded back to the 5.4, stability seemed back to normal, but now the video playback is still at 1FPS when connected to the docking station. If I unplug it, videos play at a regular rate.

Using GPU playback  from mpv:

```
[vo/gpu/opengl] EGL_VERSION=1.5
[vo/gpu/opengl] EGL_VENDOR=Mesa Project
[vo/gpu/opengl] EGL_CLIENT_APIS=OpenGL OpenGL_ES 
[vo/gpu/opengl] Trying to create Desktop OpenGL context.
[vo/gpu/opengl] Choosing visual EGL config 0x15, visual ID 0x21
[vo/gpu/opengl] GL_VERSION='4.6 (Core Profile) Mesa 20.1.2'
[vo/gpu/opengl] Detected desktop OpenGL 4.6.
[vo/gpu/opengl] GL_VENDOR='Intel'
[vo/gpu/opengl] GL_RENDERER='Mesa Intel(R) UHD Graphics 620 (WHL GT2)'
[vo/gpu/opengl] GL_SHADING_LANGUAGE_VERSION='4.60'
```

That is not restricted to mpv only, but anything that uses the GPU for videos, including Firefox on YouTube. When unplugging all works normally. Please let me know if you need more details.

**Comment by pbartyik** [comment]
Can you try routing audio to another sink (e.g. laptop's speaker)?

**Comment by pbartyik** [comment]
@UnoSD https://bbs.archlinux.org/viewtopic.php?id=256911

**Comment by UnoSD** [comment]
@pbartyik thanks. I'm using the linux-lts so it's not the same version, but I guess it may be a change applied to both. I really hope this is the case as I presume it's quicker to fix issues on the Arch/kernel side than evdi.

**Comment by displaylink-dkurek** [comment]
Please take a look at
https://www.displaylink.org/forum/showthread.php?t=67148

In short, we have raised the issue in Xserver issue tracker https://gitlab.freedesktop.org/xorg/xserver/-/issues/1028
and there is open (at the time beeing) MR with a fix
https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/448

This might be the same issue as #181

**Comment by pbartyik** [comment]
@displaylink-dkurek not sure how much this has to do with X, I'm experiencing this on wayland.

**Comment by UnoSD** [comment]
@displaylink-dkurek I should update the issue, as @pbartyik pointed out, my issue is also to do with audio only, I realised it also using and mp3 player and I have the same problem without video.

**Comment by displaylink-dkurek** [comment]
@pbartyik 
From the original message
> Xorg version (if used)
> 1.20.8-2

@UnoSD 
Sorry, missed that :)
If it's audio indeed then not sure we can do to much here, EVDI is not 
processing audio in any way. :-/

**Comment by UnoSD** [comment]
@displaylink-dkurek 
I did not think that through, I was biased as I just updated kernel and evdi and I assumed the problem was with evdi as it was happening on video playback.
Now that I know it is an audio issue, it's likely to be the kernel and/or docking station audio device.
Thank you, I'll close this issue.

**Comment by pbartyik** [comment]
@UnoSD fixed in kernel 5.7.7

**Comment by UnoSD** [comment]
thank you for the heads up @pbartyik 

I had to downgrade anyway for stability issues with the latest kernel, I will try again with the next evdi update.
